templates/repo/commit_page.tmpl

```diff
@@ -18,14 +18,125 @@
+										data-branch-from-urlcomponent="{{.CommitID}}"
+										data-modal="#create-tag-modal"
+										data-modal-from-span="#modal-create-tag-from-span"
+										data-modal-form="#create-tag-form">
+										{{.i18n.Tr "repo.tag.create_tag_operation"}}
+									</div>
+									<div class="item show-modal revert-button"
+										data-modal="#cherry-pick-modal"
+										data-modal-cherry-pick-type="revert"
+										data-modal-cherry-pick-header="{{$.i18n.Tr "repo.commit.revert-header" (ShortSha .CommitID)}}"
+										data-modal-cherry-pick-content="{{$.i18n.Tr "repo.commit.revert-content"}}"
+										data-modal-cherry-pick-submit="{{.i18n.Tr "repo.commit.revert"}}">{{.i18n.Tr "repo.commit.revert"}}</a></div>
+									<div class="item cherry-pick-button show-modal"
+										data-modal="#cherry-pick-modal"
+										data-modal-cherry-pick-type="cherry-pick"
+										data-modal-cherry-pick-header="{{$.i18n.Tr "repo.commit.cherry-pick-header" (ShortSha .CommitID)}}"
+										data-modal-cherry-pick-content="{{$.i18n.Tr "repo.commit.cherry-pick-content"}}"
+										data-modal-cherry-pick-submit="{{.i18n.Tr "repo.commit.cherry-pick"}}">{{.i18n.Tr "repo.commit.cherry-pick"}}</a></div>
+									<div class="ui basic modal" id="cherry-pick-modal">
+										<div class="ui icon header">
+											<span id="cherry-pick-header"></span>
+										</div>
+										<div class="content center">
+											<p id="cherry-pick-content" class="branch-dropdown"></p>
+											{{template "repo/branch_dropdown" dict "root" .
+												"noTag" "true" "canCreateBranch" "false"
+												"branchForm" "branch-dropdown-form"
+												"branchURLPrefix" (printf "%s/_cherrypick/%s/" $.RepoLink .CommitID) "branchURLSuffix" ""
+												"setAction" "true" "submitForm" "true"}}
+											<form method="GET" action="{{$.RepoLink}}/_cherrypick/{{.CommitID}}/{{if $.BranchName}}{{PathEscapeSegments $.BranchName}}{{else}}{{PathEscapeSegments $.Repository.DefaultBranch}}{{end}}" id="branch-dropdown-form">
+												<input type="hidden" name="ref" value="{{if $.BranchName}}{{$.BranchName}}{{else}}{{$.Repository.DefaultBranch}}{{end}}">
+												<input type="hidden" name="refType" value="branch">
+												<input type="hidden" id="cherry-pick-type" name="cherry-pick-type"><br/>
+												<button type="submit" id="cherry-pick-submit" class="ui green button"></button>
+											</form>
+										</div>
+									</div>
+									<div class="ui small modal" id="create-branch-modal">
+										<div class="header">
+											{{.i18n.Tr "repo.branch.new_branch"}}
+										</div>
+										<div class="content">
+											<form class="ui form" id="create-branch-form" action="" data-base-action="{{.RepoLink}}/branches/_new/commit/" method="post">
+												{{.CsrfTokenHtml}}
+												<div class="field">
+													<label>
+														{{.i18n.Tr "repo.branch.create_branch_operation"}}
```

Comment:
This shows "Create branch <sha>". Seems like formatting with `branch.new_branch_from` is better suited.
Same for the other modals

Reply:
Including the `Revert: <sha>` and `Cherry Pick: <sha>`? I've done the branch and tag ones.

Reply:
Nice, UI looks good to me now